cpp/memilio/io/json_serializer.h

```diff
@@ -635,23 +647,12 @@ void JsonObject::add_optional(const std::string& name, const T* value)
     }
 }
 
-template <class Iter, std::enable_if_t<JsonType<typename Iter::value_type>::value, void*>>
-void JsonObject::add_list(const std::string& name, Iter b, Iter e)
-{
-    if (m_status->is_ok()) {
-        auto& array = (m_value[name] = Json::Value(Json::arrayValue));
-        for (auto it = b; it < e; ++it) {
-            array.append(JsonType<typename Iter::value_type>::transform(*it));
-        }
-    }
-}
-
-template <class Iter, std::enable_if_t<!JsonType<typename Iter::value_type>::value, void*>>
+template <class Iter>
 void JsonObject::add_list(const std::string& name, Iter b, Iter e)
 {
     if (m_status->is_ok()) {
         auto array = Json::Value(Json::arrayValue);
-        for (auto it = b; it < e; ++it) {
+        for (auto it = b; it != e; ++it) {
```

Comment:
Less powerful iterators don't define a `<` operator, only a `!=` operator, e.g. std::list only provides bidirectional iterators instead of contiguous iterators like std::vector. So `!=` supports more containers. STL functions like `std::find` use `!=` as well.